Makefile

```diff
@@ -164,6 +164,21 @@ minor: release
 major: PATTERN = '\$$1+1\".0.0\"'
 major: release
 
+## Build and publish release using Goreleaser.
+.PHONY: publish-release
+publish-release:
+	@if ! [ -x "$$(command -v goreleaser)" ]; then \
+		echo "goreleaser is not installed, do you want to download it? [y/N] " && read ans && [ $${ans:-N} = y ]; \
+		if [ $$ans = y ] || [ $$ans = Y ]  ; then \
+			curl -sfL https://install.goreleaser.com/github.com/goreleaser/goreleaser.sh | sh; \
+		else \
+			echo "aborting make publish-release."; \
+			exit -1; \
+		fi; \
+	else \
+		goreleaser --rm-dist; \
+	fi; \
```

Comment:
We need to remove this `else` block, If I don't have `goreleaser` and install it here, I need to run `make publish-release` again to make this work.

Reply:
Oh I see. It was meant to install and run at the same time, I'll move it after the `curl | sh` bit then.

Reply:
One a second though. Might as well keep it at the end since we exit on failure.

Reply:
Yeah, definitely :)